### DISCUSSION TITLE
SUI 合约调用问题

### DISCUSSION THREAD

**Discussion by v1xingyue** [NonceGeek/Move-Camp#259] [Q&A]
链接： https://mp.weixin.qq.com/s/9PGG9G8oaPUeZYkZNjfbxg
问题范围 : 4.2.2 

$ sui client call --function buy_bread --module sandwich --package 0x08204ed92afcfdf9d0f6727a2c7d40db93a059d8 --args 0xca3f4ad2b0dea3264f38878db34cecb75a6336e0 0x520e1cecf280effc4f2129d5109f11be0c000d35 --gas [one_gas_object] --gas-budget 30000

--args 需要三个地址  
其中一个 是 Grocery 的地址，可以从部署中获取。
剩下的那两个是从哪里来的呢？

<img width="726" alt="image" src="https://user-images.githubusercontent.com/974169/215797671-cd037e50-22b1-41f3-a4d9-d0c651ac83b0.png">

**Comment by qiwihui** [comment]
`Grocery` 可以从部署过程中创建的 object 获得； 
`Coin<SUI>` 是 SUI 代币对应的 object，可以通过 `sui client gas` 命令获得，这里需要取余额大于所需的object id，如果不够，需要进行 merge； 
`TxContext` 是交易上下文，不需要传，Sui 的 move 虚拟机会自动加上；
gas object 也是 `sui client gas` 取一个满足gas要求的就可以。
